src/memory/memory.h

```diff
@@ -207,4 +213,37 @@ USE_RESULT bool memory_check_noise_remote_static_pubkey(const uint8_t* pubkey);
  */
 USE_RESULT bool memory_add_noise_remote_static_pubkey(const uint8_t* pubkey);
 
+typedef enum {
+    // success
+    MEMORY_MULTISIG_SET_BY_HASH_RESULT_OK,
+    // there was no more free slot available.
+    MEMORY_MULTISIG_SET_BY_HASH_RESULT_ERR_FULL,
+    // a different multisig with the same name already exists.
+    MEMORY_MULTISIG_SET_BY_HASH_RESULT_ERR_DUPLICATE_NAME,
+    // memory write error
+    MEMORY_MULTISIG_SET_BY_HASH_RESULT_ERR_UNKNOWN,
+} memory_multisig_set_by_hash_result_t;
+
+/**
+ * Store a multisig name under the multisig configuration ID (the hash of the config). At most
+ * MEMORY_MULTISIG_NUM_ENTRIES different configs can be stored.
+ * If a name is already stored with this hash, the old name will be overwritten.
+ * It's the callers responsibility to validate the name.
+ * @param[in] hash hash identifying the multisig config.
+ * @param[in] human readable name. Must be at most MEMORY_MULTISIG_NAME_MAX_LEN bytes,
+ * including the null terminator (otherwise the name will be truncated).
+ * @return see memory_multisig_set_by_hash_result_t.
+ */
+USE_RESULT memory_multisig_set_by_hash_result_t
+memory_multisig_set_by_hash(const uint8_t* hash, const char* name);
```

Comment:
I suspect that you can clear an entry by calling it with hash=`0xff...` and name="". Could you either document that or add a function for removing entries?

Reply:
Good catch. I am now explicitly disallowing 0xff... and empty name, and will add a deletion function in the next PR.